### ISSUE TITLE
Frida 16.4.5 Not working with Jailbroken iOS Version 15.8.2 using Dopamnie

### ISSUE THREAD

**Issue by keyurvala** [frida/frida#2916]
Hello Team,

I installed the latest Frida from the repo build.frida.re. 

When I try to execute the command `"frida-ps -U"` from a Windows PC. The device crashed and rebooted after executing the command, 

Due to the reboot, the jailbroke was also lost. I need to jailbreak the device again. Can you please help me to resolve the issue?

Thanks

**Comment by guloooo128** [comment]
same issue

**Comment by LaiKash** [comment]
Same issue here but from a MacOS. Happens in 16.4.7. I have another iOS and it doesn't work with palera1n either

**Comment by giladkr** [comment]
Same issue, I checked and it seems like 16.4.3 is working.
Not working from 16.4.4 and above

**Comment by LaiKash** [comment]
> Same issue, I checked and it seems like 16.4.3 is working. Not working from 16.4.4 and above

How do you install a lower version of the server in a jailbroken device?

**Comment by oleavr** [comment]
> Same issue, I checked and it seems like 16.4.3 is working.
> 
> Not working from 16.4.4 and above

Which side is this version on? Both or just the host?

**Comment by giladkr** [comment]
On the iphone, I downloaded `frida_16.4.3_iphoneos-arm64.deb` from the releases

**Comment by hsorbo** [comment]
Trying to repro this. 
Running ios 16.7.8 + palera1n 2.0 (rootful) and Frida 16.4.7 (both sides).
`frida-ps -U` is working fine

**Comment by giladkr** [comment]
I think its related to ios 15 devices

**Comment by LaiKash** [comment]
> I think its related to ios 15 devices

Happening with Dopanine iOS16 too. I'll recheck with palera1n in my other device

**Comment by kpratik9** [comment]
> frida_16.4.3_iphoneos-arm64.deb

Please share the link to download frida_16.4.3_iphoneos-arm64.deb

**Comment by keyurvala** [comment]
> Trying to repro this. Running ios 16.7.8 + palera1n 2.0 (rootful) and Frida 16.4.7 (both sides) macos 14.5. `frida-ps -U` is working fine

Bro is discussing Frida Version 16.4.5. The new version, 16.4.7, has just been released. It has not been checked yet. We hope it resolves the issue.

**Comment by LaiKash** [comment]
> > Trying to repro this. Running ios 16.7.8 + palera1n 2.0 (rootful) and Frida 16.4.7 (both sides) macos 14.5. `frida-ps -U` is working fine
> 
> Bro is discussing Frida Version 16.4.5. The new version, 16.4.7, has just been released. It has not been checked yet. We hope it resolves the issue.

No, it doesn't that's all about. Since 16.4.3 it is broken :) 

> Same issue here but from a MacOS and iOS 16.2. Happens in frida v16.4.7. I have another iOS and it doesn't work with palera1n either There is a workaround. Instead of using the frida-server from USB, just move it somewhere else (not /tmp) and use it with -l option over the network

**Comment by keyurvala** [comment]
I tested version Frida 16.1.6 ~ [frida_16.1.6_iphoneos-arm64.deb](https://github.com/frida/frida/releases/download/16.1.6/frida_16.1.6_iphoneos-arm64.deb)

The above version perfectly works with iOS 15.8.2 with Rootful or Rootless Device - Paleran or Dopamine. 

Step to Install:

- Install Filza File Manager
- Upload Deb File in Downloads or Temporary Location
- Install Frida-ARM64.deb file with Filza. Respring the Device.
- Run "Frida-ps -U". 

"It is working on my end. Please check yours as well."

**Comment by LaiKash** [comment]
> frida_16.4.3_iphoneos-arm64.deb

Thanks! I tried also frida_16.4.3_iphoneos-arm64.deb specified by @kpratik9 and it works too!

**Comment by kpratik9** [comment]
> pip install frida==16.1.6

Thanks mate. Your trick worked for frida_16.4.3_iphoneos-arm64.deb

@LaiKash You always welcome

**Comment by NgoHuy** [comment]
it seems to be bug with frida-tools, not frida-server

**Comment by LaiKash** [comment]
> it seems to be bug with frida-tools, not frida-server

I think it is a list of both. Latest version of frida-server makes iOS to crash but older versions (16.4.3) have issues when injecting from Darwin over USB (workaround to do it over the network `frida-server -l 0.0.0.0)

**Comment by raulsaeed** [comment]
same here with iOS 15.8 Frida version on both devices 16.4.7 iPhone just reboots after any Frida command on my Mac 

example : 
'''❯ frida-ps -Ua
Failed to enumerate applications: the connection is closed'''

**Comment by ChiChou** [comment]
@LaiKash from the panic log it seems like launchd crashed (very likely due to the agent). Can you share crash logs that named something like `launchd-1721664660.ips`?

**Comment by opa334** [comment]
Yes, please share the launchd crash the panic tells you about! Although I will try reproducing it…

**Comment by LaiKash** [comment]
@ChiChou @ChiChou Dopamine updated today and apparently specifically (and randomly) solved the crash with Frida. I checked and it is working with frida server 16.4.7 so it was apparently an issue with the frida-server.

I'll take a look to all the issues regarding this to notify but this can be closed. Thank you again!

**Comment by raulsaeed** [comment]
@LaiKash it happens to me on palera1n so it's not a dopamine problem

**Comment by LaiKash** [comment]
> @LaiKash it happens to me on palera1n so it's not a dopamine problem

Might be the same issue in both palera1n and Dopamine (which updated 3 days ago and fixed the issue without knowing how, see my previous comment)

**Comment by opa334** [comment]
Looking at the crash log, I would have said it's a frida issue aswell. But for some reason Dopamine 2.2 fixed the issue, so it might not be. I narrowed this problem down to the systemhook (not even the launchdhook, which is strange since launchd is the process that crashes). Note that palera1n uses a similar (but more minimalistic) systemhook that was adapted from Dopamine, so it's possible that that's the reason why palera1n has the bug aswell.

**Comment by keyurvala** [comment]
I updated Frida and Frida's server to the latest version and fixed the issue.

**Comment by ckgal** [comment]
I got mine working with 16.4.3 (frida_16.4.3_iphoneos-arm64.deb) for iPhone 6s; iOS 15.7.7; Dopamine iOS 15-16.5.1

on macbook
- pip install frida-tools
- pip install frida==16.4.3
 
on device
- scp frida_16.4.3_iphoneos-arm64.deb mobile@<ip addr>/tmp
- dpkg -i frida_16.4.3_iphoneos-arm64.deb

**Comment by opa334** [comment]
> I got mine working with 16.4.3 (frida_16.4.3_iphoneos-arm64.deb) for iPhone 6s; iOS 15.7.7; Dopamine iOS 15-16.5.1
> 
> on macbook
> 
> * pip install frida-tools
> * pip install frida==16.4.3
> 
> on device
> 
> * scp frida_16.4.3_iphoneos-arm64.deb mobile@/tmp
> * dpkg -i frida_16.4.3_iphoneos-arm64.deb

all you need to do is use the latest version of Dopamine, the issue is fixed
